### PR TITLE
Redirect Get message by id request when broker not serve for the topic

### LINKED ISSUE #7604: [Rest admin api] 'Get message by its messageId' throw exception when the broker asked is not the good one
**Describe the bug**

When using the rest api `http://pulsar.apache.org/admin-rest-api/#operation/getLastMessageId` you may have exception if the broker asked is not the good one. By executing the same request several times, it will succeeded with the right answer few times or it will fail with this exception.

For exemple the request I used :
`curl --location --request GET 'https://host:2000/admin/v2/persistent/mytenant/myns/mytopic/ledger/757808/entry/0' `

 ```
--- An unexpected error occurred in the server ---

Message: org.apache.pulsar.broker.service.BrokerServiceException$ServiceUnitNotReadyException: Namespace bundle for topic (persistent://mytenant/myns/mytopic) not served by this instance. Please redo the lookup. Request is denied: namespace=mytenant/myns

Stacktrace:

java.util.concurrent.ExecutionException: org.apache.pulsar.broker.service.BrokerServiceException$ServiceUnitNotReadyException: Namespace bundle for topic (persistent://mytenant/myns/mytopic) not served by this instance. Please redo the lookup. Request is denied: namespace=mytenant/myns
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1915)
	at org.apache.pulsar.broker.admin.impl.PersistentTopicsBase.getTopicReference(PersistentTopicsBase.java:2353)
	at org.apache.pulsar.broker.admin.impl.PersistentTopicsBase.internalGetMessageById(PersistentTopicsBase.java:1836)
	at org.apache.pulsar.broker.admin.v2.PersistentTopics.getMessageById(PersistentTopics.java:956)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
....
Caused by: org.apache.pulsar.broker.service.BrokerServiceException$ServiceUnitNotReadyException: Namespace bundle for topic (persistent://mytenant/myns/mytopic) not served by this instance. Please redo the lookup. Request is denied: namespace=mytenant/myns
	at org.apache.pulsar.broker.service.BrokerService.checkTopicNsOwnership(BrokerService.java:1372)
	at org.apache.pulsar.broker.service.BrokerService.loadOrCreatePersistentTopic(BrokerService.java:955)
	at org.apache.pulsar.broker.service.BrokerService.lambda$getTopic$13(BrokerService.java:732)
	at org.apache.pulsar.common.util.collections.ConcurrentOpenHashMap$Section.put(ConcurrentOpenHashMap.java:277)
	at org.apache.pulsar.common.util.collections.ConcurrentOpenHashMap.computeIfAbsent(ConcurrentOpenHashMap.java:130)
	at org.apache.pulsar.broker.service.BrokerService.getTopic(BrokerService.java:731)
	at org.apache.pulsar.broker.service.BrokerService.getTopicIfExists(BrokerService.java:711)
	at org.apache.pulsar.broker.admin.impl.PersistentTopicsBase.getTopicReference(PersistentTopicsBase.java:2352)
	... 73 more

```

**Expected behavior**
The request should always succeed and the request should be forwarded to the right broker.

### PR DESCRIPTION
**Pull request by rudy2steiner**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #7604

 

Master Issue: #7604

### Motivation

We should redirect Get message by id request when a broker doesn't serve for the topic

### Modifications

* Redirect Get message by id request when a broker doesn't serve for the topic
* ~~Test log config use log4j2.xml not logback.xml
     config by parent pom: -Dlog4j.configurationFile=log4j2.xml~~

### Verifying this change

This change is a trivial changes without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API: (yes )
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: ( no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### CODE REVIEW

File: pulsar-broker/src/test/resources/log4j2.xml [hunk 1]
```diff
@@ -19,18 +19,16 @@
 
 -->
 <configuration scan="true">
-<!--
```
Comment (sijie) [inline pulsar-broker/src/test/resources/log4j2.xml:22]: The change doesn't seem to be related.

Comment (rudy2steiner) [reply pulsar-broker/src/test/resources/log4j2.xml:22]: When I try to debug the API, using LoadBalancerTest to launch a multi nodes cluster and trying to change the debug level, such as org.apache.pulsar.broker.admin. but logback.xml configuration doesn't work and found that we actually using log4j2.xml which is configured in parent pom.

I just change the log config file by the way, it's ok if you still think we shouldn't change this

Comment (rudy2steiner) [reply pulsar-broker/src/test/resources/log4j2.xml]: revert

**Comment by rudy2steiner** [comment]
@sijie  please help review !

**Comment by rudy2steiner** [comment]
I will check the unit test exception,seem like not cause by this changes:
```
org.testng.internal.thread.ThreadTimeoutException: Method org.apache.pulsar.client.api.SimpleProducerConsumerTest.testSharedSamePriorityConsumer() didn't finish within the time-out 5000
```

**Comment by rudy2steiner** [comment]
I try to run the same unit test command on local
```
./build/retry.sh mvn -B -ntp test -pl pulsar-broker -Dinclude='org/apache/pulsar/broker/**/*.java' -Dexclude='org/apache/pulsar/broker/zookeeper/**/*.java,org/apache/pulsar/broker/loadbalance/**/*.java,org/apache/pulsar/broker/service/**/*.java,**/AdminApiOffloadTest.java'
```
without any failed test case.
But online test failed after retry 3 times and each time failed with different test cases:
```
[ERROR] Failures: 
[ERROR] org.apache.pulsar.broker.transaction.coordinator.TransactionMetaStoreAssignmentTest.testTransactionMetaStoreAssignAndFailover(org.apache.pulsar.broker.transaction.coordinator.TransactionMetaStoreAssignmentTest)
[INFO]   Run 1: PASS
[ERROR]   Run 2: TransactionMetaStoreAssignmentTest.testTransactionMetaStoreAssignAndFailover:45 expected [16] but found [0]
[INFO] 
[INFO]
-----------
[ERROR] Failures: 
[ERROR] org.apache.pulsar.broker.admin.IncrementPartitionsTest.testIncrementPartitionsWithNoSubscriptions(org.apache.pulsar.broker.admin.IncrementPartitionsTest)
[INFO]   Run 1: PASS
[ERROR]   Run 2: IncrementPartitionsTest.testIncrementPartitionsWithNoSubscriptions:130 expected [20] but found [10]
[INFO] 
[ERROR]   TransactionProduceTest.setup ? PulsarAdmin java.util.concurrent.CompletionExce...
[INFO] 
[ERROR] Tests run: 417, Failures: 2, Errors: 0, Skipped: 2

-----
[ERROR] Failures: 
[ERROR] org.apache.pulsar.broker.admin.IncrementPartitionsTest.testIncrementPartitionsWithNoSubscriptions(org.apache.pulsar.broker.admin.IncrementPartitionsTest)
[INFO]   Run 1: PASS
[ERROR]   Run 2: IncrementPartitionsTest.testIncrementPartitionsWithNoSubscriptions:127 expected [10] but found [2]
[INFO] 
[ERROR] org.apache.pulsar.broker.admin.MaxUnackedMessagesTest.testMaxUnackedMessagesOnSubscriptionApi(org.apache.pulsar.broker.admin.MaxUnackedMessagesTest)
[INFO]   Run 1: PASS
[ERROR]   Run 2: MaxUnackedMessagesTest.testMaxUnackedMessagesOnSubscriptionApi ? ServerSideError
[INFO] 
[ERROR] org.apache.pulsar.broker.transaction.coordinator.TransactionMetaStoreAssignmentTest.testTransactionMetaStoreAssignAndFailover(org.apache.pulsar.broker.transaction.coordinator.TransactionMetaStoreAssignmentTest)
[INFO]   Run 1: PASS
[ERROR]   Run 2: TransactionMetaStoreAssignmentTest.testTransactionMetaStoreAssignAndFailover:45 expected [16] but found [0]
[INFO] 
[INFO] 
[ERROR] Tests run: 415, Failures: 3, Errors: 0, Skipped: 0
```
